src/main/java/tech/fastj/engine/internals/ThreadFixer.java

```diff
@@ -27,7 +27,7 @@ public static void start() {
                     try {
                         Thread.sleep(Long.MAX_VALUE);
                     } catch (InterruptedException exception) {
-                        FastJEngine.warning("Interrupted while in daemon sleep: " + exception.getMessage() + Arrays.deepToString(exception.getStackTrace()));
+                        FastJEngine.warning("Interrupted while in daemon sleep: {}" + exception.getMessage() + Arrays.deepToString(exception.getStackTrace()));
```

Comment:
Seems like you've added the {} formatting, but the string concatenation is still present here.